core/consensus/beefy/impl/beefy_impl.cpp

```diff
@@ -393,9 +397,53 @@ namespace kagome::network {
     return outcome::success();
   }
 
+  outcome::result<void> BeefyImpl::fetchHeaders() {
+    std::unordered_map<primitives::BlockNumber, BlockDataToStore> blocksToStore;
+    while (true) {
+      auto& fetchingHeader = *fetching_headers_;
+      auto block_hash = block_tree_->getBlockHash(fetchingHeader);
+      if (block_hash) {
+        auto& blockHashOptValue = block_hash.value();
+        if (blockHashOptValue.has_value()) {
+          const auto& hashValue = *blockHashOptValue;
+          auto blockHeader = block_tree_->getBlockHeader(hashValue);
+          if (blockHeader) {
+            const auto& blockHeaderValue = blockHeader.value();
+            blocksToStore[fetchingHeader] = {
+              .header = blockHeaderValue,
+              .hash = hashValue
+            };
+            if (beefyValidatorsDigest(blockHeaderValue).has_value()) {
+              beefy_justification_protocol_.get()->fetchJustification(fetchingHeader);
+              break;
```

Comment:
There may be no justification, so we should continue.
```suggestion
```

Reply:
And what's the break condition then?

Reply:
missing block breaks loop to fetch,
`*fetching_headers_ <= beefy_genesis_` stops search completely